src/spring.js

```diff
@@ -0,0 +1,54 @@
+import { createEndValueType, FRAME_RATE, noSpeed, updateCurrV, updateCurrVals } from './utils';
+import Motion from './motion';
+import { PropTypes } from 'react';
+
+export default class Spring extends Motion {
```

Comment:
Common code can be extracted to functions. If it cannot be extracted, then core working component can be wrapped into two main components. Not inherited.

Considering most of the functionality is related to pure calc, it should be very easy to extract common code to pure functions.